include/odp/api/spec/packet.h

```diff
@@ -388,6 +416,29 @@ odp_event_t odp_packet_to_event(odp_packet_t pkt);
 void odp_packet_to_event_multi(const odp_packet_t pkt[], odp_event_t ev[],
 			       int num);
 
+/**
+ * Get partial reassembly state from a packet
+ *
+ * In case of incomplete reassembly, a packet carries information on
+ * the time already used for the reassembly attempt and one or more
+ * fragments. The fragments are not necessarily the original received
+ * fragments but may be partially reassembled parts of the packet.
+ *
+ * This function may be called only if the reassembly status of a packet
+ * is ODP_PACKET_REASS_INCOMPLETE.
+ *
+ * @param      pkt   Incompletely reassembled packet. The packet will
+ *                   be consumed if the function succeeds.
+ * @param[out] frags Packet handle array for output. The size of this array must
+ *                   be at least `odp_reass_capability.max_num_frags`.
```

Comment:
Maybe we should have max_num_frags as a config parameter too to make it possible for an application to limit the number of fragments it may get from odp_packet_reass_partial_state() and to limit the resources that might need to be reserved in the implementation.

Or as a possible alternative, make odp_packet_reass_partial_stat_t.num_frags work as an input parameter too, telling the maximum number of fragments that can be returned. In that case the returned num_frags could still tell the total number of fragments even though not all were written in the packet array (and then maybe the function should not consume the REASS_INCOMPLETE packet to allow retry with a larger array).

Reply:
> 
> 
> Maybe we should have max_num_frags as a config parameter too to make it possible for an application to limit the number of fragments it may get from odp_packet_reass_partial_state() and to limit the resources that might need to be reserved in the implementation.

Agreed. This was in the original proposal. I had dropped in the recent version when we decided to decouple capability & config. But I see the advantage in having this in config.

> 
> Or as a possible alternative, make odp_packet_reass_partial_stat_t.num_frags work as an input parameter too, telling the maximum number of fragments that can be returned. In that case the returned num_frags could still tell the total number of fragments even though not all were written in the packet array (and then maybe the function should not consume the REASS_INCOMPLETE packet to allow retry with a larger array).

Well, then it would be datapath checks which can be easily avoided with a config that application should be expected to honor. Also, the number of fragments that can be reassembled may not be that large to require the need for such flexibility. So I would prefer the first option.

Reply:
> So I would prefer the first option.

So would I. I just wanted to mention an alternative.

Reply:
This line (the description of frags) should now refer to odp_reass_config_t::max_num_frags.

Reply:
Missed this. Updated in v8.